pkg/registry/generic/registry/store.go

```diff
@@ -540,7 +567,12 @@ func markAsDeleting(obj runtime.Object) (err error) {
 	return nil
 }
 
-// this functions need to be kept synced with updateForGracefulDeletionAndFinalizers.
+// These functions needs to be kept synced:
+//
+//   updateForGracefulDeletion
+//   updateForGracefulDeletionAndFinalizers
+
+// updateForGracefulDeletion
```

Comment:
yep, this is still WIP.  I have to digest those both a teensy bit more before i can articulate a good comment.